pandas/tests/indexes/common.py

```diff
@@ -937,3 +937,14 @@ def test_empty(self):
         index = self.create_index()
         assert not index.empty
         assert index[:0].empty
+
+    @pytest.mark.parametrize('how', ['outer', 'inner', 'left', 'right'])
+    def test_join_self(self, how):
+        index = self.create_index()
+        joined = index.join(index, how=how)
+
+        if index is index.unique():
```

Comment:
yes some are not unique and the result is some modified cartesian product of the two so the test was failing for that case.  So i modified the test to check that at least an index of the same type was returned (potentially a more rigorous test should be done maybe in`tests/indexes/period.py`?).